pytorch_lightning/core/lightning.py

```diff
@@ -169,6 +168,10 @@ def automatic_optimization(self) -> bool:
         """
         return self._automatic_optimization
 
+    @property
+    def running_stage(self):
```

Comment:
return value

Reply:
I'm sorry, I don't follow this or the other comment you made. Can you reformulate?

Reply:
Sorry!

> return value

Can you add a return typing annotation for this function, please? 😄 

> Does the model not get the attribute with these changes?

By looking at the before changes, I see `_set_runinng_stage` set also `model_ref.running_stage = stage`. Which I assume meant one could do `model.running_stage`. Is that no longer the case?

Reply:
model.running_stage is now a property (read-only). It should be like this from the beginning, as there is no real reason why the user would ever change that, and besides it should always be the same value as the property in the trainer. We have the same logic for other attributes like global_rank, current_epoch etc.

Reply:
Also, there is no valid "running_stage" available when the model is not attached to a Trainer.

Reply:
@carmocca I can't add that type hint, it gives me cyclic import errors. Looks hopeless .(

Reply:
even with `typing.TYPE_CHECKING`?

circular imports is the worst part about python lol

Reply:
I will try that but that's probably only going to work on new python versions and then still fail in our ci, let's see

Reply:
@awaelchli  you could import RunningStage guarded bu `with typing.TYPE_CHECKING` as @carmocca mentioned, but you need to change the annotation to `-> Optional['RunningStage']`. This name will be resolved to the imported RunningStage during type checking and be a string otherwise, so no issue with circular imports or undefined things :)